crates/matrix-sdk-ui/src/timeline/builder.rs

```diff
@@ -265,22 +263,9 @@ impl TimelineBuilder {
                         RoomEventCacheUpdate::AddTimelineEvents { events, origin } => {
                             trace!("Received new timeline events.");
 
-                            // Special case for pinned events: when we receive new events what
-                            // we'll do is, instead of adding the events, update the pinned events
-                            // cache with them, reload the list of pinned event ids and reload the
-                            // list of pinned events with this info.
-
-                            // TODO(bnjbvr): why is this done? the timeline already listens to
-                            // changes to the pinned events list, so we shouldn't have to trigger a
-                            // manual update here too.
-                            if let TimelineFocus::PinnedEvents { .. } = &*focus.clone() {
-                                if let Some(ret) = inner.update_pinned_events_if_needed(events, &pinned_event_cache).await {
-                                    match ret {
-                                        Ok(events) => inner.replace_with_initial_remote_events(events, RemoteEventOrigin::Sync).await,
-                                        Err(err) => error!("Couldn't update pinned events with incoming timeline events: {err}"),
-                                    }
-                                }
-                            } else {
+                            // Note: we deliberately choose to not handle
+                            // updates/reactions/redactions for pinned events.
```

Comment:
This is temporary, right? We do want to know about updates/redactions. Or does it just mean we don't handle them here?

Reply:
This was the previous behavior: the new events were not added, so I assumed this was expected. (At least it wasn't tested since no test failed from this change.)

Reply:
Unless I missed something [the very first test](https://github.com/matrix-org/matrix-rust-sdk/blob/1c4f035c99284e6c245843ec0066a4e786726a37/crates/matrix-sdk-ui/tests/integration/timeline/pinned_event.rs#L22) should handle that, I'm not sure why there was no behaviour change 🫤 .

Reply:
Ok, so this was a misunderstanding on my side of how the Matrix spec works. I thought for updates and redaction I'd get an event with the same event id but different contents. Like this, it would be filtered, I'd update the cache with the new contents, then reload the list taking this value from the cache.

But if it doesn't work like that, then I wasn't really doing anything.